pkg/operator/controller/dns_status_test.go

```diff
@@ -212,142 +225,204 @@ func TestComputeDNSDegradedCondition(t *testing.T) {
 			},
 		}
 	}
+
+	defaultOldCondition := &operatorv1.OperatorCondition{
+		Type:   operatorv1.OperatorStatusTypeDegraded,
+		Status: operatorv1.ConditionUnknown,
+	}
+
+	defaultOldGraceCondition := &operatorv1.OperatorCondition{
+		Type:   DNSMaxUnavailableDNSPodsExceeded,
+		Status: operatorv1.ConditionUnknown,
+	}
```

Comment:
The omission of `LastTransitionTime` is conspicuous, but the test works because both the fake clock as well as the last transition time of these status conditions are initialized with time 0.  It would be safer though to set the last transition times using the fake clock:
```suggestion
	defaultOldCondition := &operatorv1.OperatorCondition{
		Type:               operatorv1.OperatorStatusTypeDegraded,
		Status:             operatorv1.ConditionUnknown,
		LastTransitionTime: metav1.NewTime(clock.Now()),
	}

	defaultOldGraceCondition := &operatorv1.OperatorCondition{
		Type:               DNSMaxUnavailableDNSPodsExceeded,
		Status:             operatorv1.ConditionUnknown,
		LastTransitionTime: metav1.NewTime(clock.Now()),
	}
```